src/app/login/form.tsx

```diff
@@ -21,13 +24,71 @@ export default function Form({ initialEmail }: { initialEmail: string | null })
     const buttonDisabled = disabled || pending
     const handleSubmit: MouseEventHandler = async (e) => {
         setPending(true)
-        await signIn('email', { email })
+        const res = await signIn('email', { email, redirect: false })
         setPending(false)
+        if (res?.error) {
+            if (res?.url) {
+                router.replace(res.url);
+            } else {
+                router.replace(`/login?error=${encodeURIComponent(res.error)}`)
+            }
+        } else {
+            router.replace('/login')
+            onSuccess(email);
+        }
     }
-    return <form>
+    return <form onSubmit={(e) => e.preventDefault()}>
         <div className={styles.formSection}>
             <input id="email" name="email" type="email" placeholder="Email" className={styles.textInput} required={true} value={email} onChange={e => setEmail(e.target.value)} />
-            <Button disabled={buttonDisabled} pending={pending} label="Send me a link" pendingLabel="Sending..." role="success" onClick={handleSubmit} />
+            <Button disabled={buttonDisabled} pending={pending} label="Send me a code" pendingLabel="Sending..." role="success" onClick={handleSubmit} />
+        </div>
+        <Subtitle>Tally will email you a code to create your account or login to your existing account.</Subtitle>
+    </form>
+}
+
+const VerifyCode = ({ email }: { email: string }) => {
+    const [code, setCode] = React.useState('')
+    const [disabled, setDisabled] = React.useState(true)
+    const router = useRouter()
+
+    React.useEffect(() => {
+        setDisabled(!code)
+    }, [code])
+
+    const handleSubmit = async () => {
+        const res = await fetch(`/api/auth/callback/email?email=${encodeURIComponent(email)}&token=${encodeURIComponent(code)}&callbackUrl=${encodeURIComponent('/login-success')}`)
```

Comment:
navigating directly here leads to terrible multiple refreshes before landing on eventually correct state. just detecting this dummy `login-success` value in the response is a hack but it works. we refresh once and the login page redirects us to the user page, all in one navigation.